File: src/models/output/disaggregated_vars.jl
function gen_disaggregated_var_df(m)

    var = [
        "x_load_D", "x_fuel_D", "x_CO2_D_flue",
        "x_power_D_HP", "x_power_D_IP", "x_power_D_aux",
        "x_steam_D_DAC_base", "x_steam_D_allocable"
    ]

    l = []
    for v in var
        l = vcat(l, [v * string(mod) => Float64[] 
            for mod in set_mode])
    end

    df_disaggregated_vars = DataFrame(l)

    for i in set_hour
        local l0 = []
        for v in var
            s = Symbol(v)
            local l = [value(m[s][i, mod]) for mod in set_mode]
            l0 = vcat(l0, l)
        end
        push!(df_disaggregated_vars, l0)
    end

    return df_disaggregated_vars

end

File: src/pyomo_model/output/disaggregated_vars.py
"""
Output module
Pengfei Cheng

Output disaggregated variable-related results.
"""

import pandas as pd
from pyomo.environ import value
from src.pyomo_model.params import *


def gen_disaggregated_var_df(m, set_hour):

    vars = [
        "x_load_D", "x_fuel_D", "x_CO2_D_flue",
        "x_power_D_HP", "x_power_D_IP", "x_power_D_aux",
        "x_steam_D_DAC_base", "x_steam_D_allocable"
    ]

    dfs = {}

    for s in set_scenario:
        _dict = {}
        for var_name in vars:
            _tmp_dict = {}
            for i in set_hour:
                for j in set_mode:
                    var = getattr(m, var_name)
                    _tmp_dict[i, j] = value(var[i, j, s])
            _dict[var_name] = _tmp_dict

        dfs[s] = pd.DataFrame.from_dict(_dict)

    return dfs